jenkins/format-submission.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import requests
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import HTTPError
+import subprocess
+import zipfile
+
+# submission process starts from MICROSUB project on issues.citrite.net
+HOST = "https://issues.citrite.net"
+
+# Submission issue type has multiple custom fields
+SUPPORT_URL = "customfield_12635"
+DOCUMENT_URL = "customfield_31030"
+PRIVACY_URL = "customfield_31032"
+TERMS_URL = "customfield_31031"
+VENDOR = "customfield_31230"
+
+# metadata for bundle is held in the METADATA_FILE
+METADATA_FILE = "metadata.json"
+
+# parse args to obtain credentials for Jira API
+parser = argparse.ArgumentParser(description='Format the export to include extra metadata from third parties')
+parser.add_argument('--issueId', type=str, required=True, help='id of the issue that triggered the build')
+parser.add_argument('--svcacctPwd', type=str, required=True, help='password for the service account making call to Jira')
+parser.add_argument('--svcacctName', type=str, required=True, help='service account name used for Jira API calls')
+args = parser.parse_args()
+SVCACCT_NAME = args.svcacctName
+SVCACCT_PWD = args.svcacctPwd
+ISSUE_ID = args.issueId
+AUTH = HTTPBasicAuth(SVCACCT_NAME, SVCACCT_PWD)
+HEADERS = {'Content-Type': 'application/json'}
+
+
+def _error_writeback(error):
+    errorMessage = "Jenkins script failed: " + error
+    payload = json.dumps( {
+        "body": errorMessage
+    } )
+    try:
+        response = requests.post(
+            url='{}/rest/api/2/issue/{}/comment'.format(HOST, ISSUE_ID),
+            headers=HEADERS,
+            auth=AUTH,
+            data=payload
+        )
+    except:
+        print(response.text)
+    exit(1)
+
+
+def download_attachments():
+    params = {'fields': 'attachment'}
+    attachments = []
+    try:
+        response = requests.get(
+            url='{}/rest/api/2/issue/{}'.format(HOST, ISSUE_ID),
+            headers=HEADERS,
+            auth=AUTH,
+            params=params
+        )
+        jsonData = response.json()
+        for attachment in jsonData['fields']['attachment']:
```

Comment:
Do we need to download every attachment? Couldn't we just download the one that ends in `*.mapp`?

Reply:
I see that you already have this logic in the "find_mapp_file" function, we should apply it here before we have done the download.